src/pint/utils.py

```diff
@@ -1580,21 +1620,246 @@ def companion_mass(pb, x, inc=60.0 * u.deg, mpsr=1.4 * u.solMass):
     return x1.to(u.Msun)
 
 
-def ELL1_check(A1, E, TRES, NTOA, outstring=True):
+@u.quantity_input
+def PBDOT(mp: u.Msun, mc: u.Msun, pb: u.d, e: u.dimensionless_unscaled):
+    """Post-Keplerian orbital decay PBDOT, assuming general relativity.
+
+    PBDOT (:math:`\dot P_B`) is the change in the binary orbital period
+    due to emission of gravitational waves.
+
+    Parameters
+    ----------
+    mp : astropy.units.Quantity
+        pulsar mass
+    mc : astropy.units.Quantity
+        companion mass
+    pb : astropy.units.Quantity
+        Binary orbital period
+    e : Quantity or float
+        orbital eccentricity
+
+    Returns
+    -------
+    pbdot : astropy.units.Quantity
+        (dimensionless)
+
+    Raises
+    ------
+    astropy.units.UnitsError
+        If the input data are not appropriate quantities
+    TypeError
+        If the input data are not quantities
+
+    Notes
+    -----
+    Calculates
+
+    .. math::
+        \dot P_b = -\\frac{192\pi}{5}T_{\odot}^{5/3} \\left(\\frac{P_b}{2\pi}\\right)^{-5/3}
+        f(e)\\frac{m_p m_c}{(m_p+m_c)^{1/3}}
+        
+    with
+
+    .. math::
+        f(e)=\\frac{1+(73/24)e^2+(37/96)e^4}{(1-e^2)^{7/2}}
+
+    and :math:`T_\odot = GM_\odot c^{-3}`.
+
+    See [1]_
+
+    .. [1] Lorimer & Kramer, 2008, "The Handbook of Pulsar Astronomy", Eqn. 8.52
+
+    """
+    f = (1 + (73.0 / 24) * e ** 2 + (37.0 / 96) * e ** 4) / (1 - e ** 2) ** (7.0 / 2)
+    value = (
+        (const.G / const.c ** 3) ** (5.0 / 3)
+        * (pb / (2 * np.pi)) ** (-5.0 / 3)
+        * (-192 * np.pi / 5)
+        * f
+        * (mp * mc)
+        / (mp + mc) ** (1.0 / 3)
+    )
+    return value.to(u.s / u.s)
+
+
+@u.quantity_input
+def GAMMA(mp: u.Msun, mc: u.Msun, pb: u.d, e: u.dimensionless_unscaled):
```

Comment:
Do these behave okay with arrays? I'm guessing they just broadcast each thing against all the others and then it works, but it might be good to check it in the test cases and then say so.

Reply:
What are you asking particularly?  The function or the `quantity_input`?  Using `quantity_input` works.  I just added some explicitly array-based tests for the `A1SINI` function that also uses `companion_mass` or `pulsar_mass`.  And the function itself is just math.  Is there some aspect you are suspicious of?

Reply:
No, I was pretty sure they ran fine with arrays - but when implementing a heap of Keplerian stuff, I found that converting between eccentric, mean, and true anomalies needed an iterative process that prevented functions from working array-wise. So it's helpful if functions have a short note to indicate whether they work array-wise to save the user guessing.

Reply:
OK, can do.  Since these are direct calculations there is no difficulty (but I have had similar experiences with the various anomalies).

Reply:
The right answer for the anomalies is probably either to explicitly write an array-friendly version or use `np.vectorize`. I've tried to build a third approach where functions only ever have to deal with a 1D array and the decorator handles all the broadcasting and reshaping but I never came up with anything really satisfactory.